targets/TARGET_ARM_SSG/TARGET_MUSCA_B1/CMakeLists.txt

```diff
@@ -4,6 +4,15 @@
 if("TFM" IN_LIST MBED_TARGET_LABELS)
     include(mbed_set_post_build_tfm)
 
+    # Check deps for image signing
+    if(NOT(Python3_FOUND AND HAVE_PYTHON_INTELHEX AND HAVE_PYTHON_CBOR AND
+            HAVE_PYTHON_CLICK AND HAVE_PYTHON_CRYPTOGRAPHY))
+        message(FATAL_ERROR "Missing Python dependencies (python3, cbor, "
+        "click, intelhex) so the binary cannot be signed. Please install "
+        "requirements with:\n"
+        "\tpython3 -m pip install -r mbed-os/requirements.txt")
```

Comment:
We initially wanted to separate the deps between the old tools and CMake. 
This should point to `https://github.com/ARMmbed/mbed-os/blob/master/tools/cmake/requirements.txt` instead? 

Not all this required packages are there listed?

Reply:
mbed-os/requirements.txt are needed, not just tools/cmake/requirements.txt, because of the signing tools. Maybe the right solution is to add the dependencies needed for the post build hooks to tools/cmake/requirements.txt and point there.

Reply:
How do we want to handle optional dependencies? Another requirements.txt? We have if statements in other cmake files to avoid printing the map file summary if intelhex is not installed, for example. However, intelhex is required for code signing on certain targets.

Reply:
> Maybe the right solution is to add the dependencies needed for the post build hooks to tools/cmake/requirements.txt and point there.

+1, they are missing there once we ported post build hooks 😞 I can create an issue for this, we should fix it.

> How do we want to handle optional dependencies? 

Previous mbed way it was just add to to requirements and have it installed. Shall we split requirements. We always discussed we have different needs:

requirements - to install all below (for CI, or users who want to just install everything )
requirements-basic - this would be basic requirements for targets without any post build hooks
requirements-testing - testing (build and run together in one) - including unittest/greentea tests etc?
requiremetns-<vendor> - I assume going down to Mbed OS target would be too many files so rather each vendor would maintain  their own requirements and keep it up to date. But maybe this is not needed and just introduce requirements-optional to contain all these. We can always check with CMake if deps are installed and if not , just point to extra requirements (in this case we know a user did not install all requirements so they need to cherry-pick extra or testing explicitely)

Reply:
All the requirements files seems messy to me. It's more stuff for a user to have to read and learn about. I'll add the optional requirements to tools/cmake/requirements.txt for now, even if not all targets need them.